src/Confluent.Kafka.Avro/GenericSerializerImpl.cs

```diff
@@ -105,29 +83,30 @@ public byte[] Serialize(string topic, GenericRecord data)
                 // slow since writerSchemaString is potentially long. It would be
                 // better to use hash functions based on the writerSchemaString 
                 // object reference, not value.
+
+                writerSchema = data.Schema;
+
                 string subject = this.isKey
                     ? schemaRegistryClient.ConstructKeySubjectName(topic)
                     : schemaRegistryClient.ConstructValueSubjectName(topic);
-                var subjectSchemaPair = new KeyValuePair<string, string>(subject, writerSchemaString);
-                if (!registeredSchemas.Contains(subjectSchemaPair))
+  
+                if (!registeredSchemas.ContainsKey(subject))
```

Comment:
thanks for the PR @NikolayDekov! There's a problem here though - if `data.Schema` changes between calls, the wrong schema id will be prepended to the message.